### DISCUSSION TITLE
Pgwatch3 push mode

### DISCUSSION THREAD

**Discussion by ondar** [cybertec-postgresql/pgwatch#549] [Q&A] [answered]
I'm trying to find out a good way to configure pgwatch3 for our network setup. Is it possible to run locally-configured (env, file, cmdline args, ...) next to every monitored DB but store metrics results to 1 global DB instance?

**Comment by pashagolub** [comment, marked as answer]
Yes, it's possible.

Just run as many pgwatch instances as you want but specify the same sink for metrics:

```
./pgwatch3 --config=<env var, local file, db>  --sink=postgresql://pgwatch3@targethost/pgwatch3_metrics
```

in pgwatch3 you may specify as many parallel sinks as you want, thus you can have two copies of measurements, e.g. one local and one in a global database:

```
./pgwatch3 ...  --sink=jsonfile://metrics.json  --sink=postgresql://pgwatch3@localhost/pgwatch3_metrics --sink=prometheus://:9127
```

**Reply by ondar** [reply]
Thanks, I will try that.

**Reply by ondar** [reply]
@pashagolub Is it also possible to select which metrics to go to which sink (like to have only some of them to go to global DB) and is there any buffering or retry mechanism when this global DB is temporarily unreachable?

**Reply by pashagolub** [reply]
Nope, sinks are per instance. So another solution would be to run separate pg_watch instances per sink.
And you can have as many sources for the same database as you want but with different set of metrics. So for example, one pgwatch monitors some global metrics and measurements go to some global db, another one some frequent metrics and measurements go to local one with shorter retention.

Yes, there is retry mechanism for database and file sinks, for prometheus pgwatch always serves the latest measurements.
